### DISCUSSION TITLE
Using web component libraries in Solid

### DISCUSSION THREAD

**Discussion by daron81** [solidjs/solid#558] [Q&A] [answered]
Given that solid-js has partially synthetic events, it possible to use web component ui libraries in solid-js directly, or do they require a wrapper like React does?

**Comment by ryansolid** [comment, marked as answer]
It should be fine. I already handle event retargetting. I also have some conventions. Although you write attributes on web component JSX like attributes it automatically maps to camelCase props by default and passes rich values. If you want to use the attribute form explicitly or set using a specific prop use the `attr:` and `prop:` namespace prefixes to have them process.

**Reply by daron81** [reply]
Thanks, Ryan. This site has some tests for compatibility with various frameworks: https://custom-elements-everywhere.com/

However, the solid-js tests are failing for some reason.

**Reply by ryansolid** [reply]
Solid passes all those tests. It is broken because they updated Solid after over a year and I had introduced breaking changes. 

I've had an open PR for a few months now, but the repo isn't really actively maintained.
https://github.com/webcomponents/custom-elements-everywhere/pull/1005

For reference the PR adding Solid to that repo took 2 years to merge.
